syntax = "proto3";

import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

package analyzer.v1;

message Status {
  int32 code = 1;
  string message = 2;
  repeated google.protobuf.Any details = 3;
}

message Location {
  string id = 1;
  float latitude = 2;
  float longitude = 3;
}

message Recording {
  string id = 1;
  float duration = 2;
  google.protobuf.Timestamp start_time = 3;
  google.protobuf.Timestamp end_time = 4;
  string storage_uri = 5;
}

message Detection {
  float start_time = 1;
  float end_time = 2;
  float confidence = 3;
  string common_name = 4;
  string scientific_name = 5;
  string label = 6;
}

message AnalyzeRequest {
  Recording recording = 1;
  Location location = 2;
}

message AnalyzeResponse {
  Status status = 1;
  repeated Detection detections = 2;
}

service AnalyzerService {
  rpc Analyze (AnalyzeRequest) returns (AnalyzeResponse) {}
}
